astronomer/providers/snowflake/extractors/snowflake.py

```diff
@@ -15,9 +15,9 @@ class SnowflakeAsyncExtractor(BaseExtractor):
     source_type = "SNOWFLAKE"
     default_schema = "PUBLIC"
 
-    def __init__(self, operator):  # type: ignore[no-untyped-def]
+    def __init__(self, operator: BaseOperator) -> None:
         super().__init__(operator)
-        self.conn: "Connection" = None
+        self.conn: Connection = Connection()
```

Comment:
No,  we shouldn't need to do this

Reply:
mypy fails otherwise
```
astronomer/providers/snowflake/extractors/snowflake.py: note: In member "__init__" of class "SnowflakeAsyncExtractor":
astronomer/providers/snowflake/extractors/snowflake.py:20: error: Incompatible
types in assignment (expression has type "None", variable has type "Connection")
 [assignment]
            self.conn: "Connection" = None
```

Reply:
So you should change it to: `self.conn: Connection | None = None`

Reply:
as we don't want to instantiate it unncessarily